modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQJMSContext.java

```diff
@@ -0,0 +1,729 @@
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() {
+        try {
+            connection.start();
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        try {
+            connection.stop();
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setAutoStart(boolean autoStart) {
+        this.autoStart = autoStart;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean getAutoStart() {
+        return autoStart;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void close() {
+        boolean isExceptionThrown = false;
+        try {
+            if (session != null) {
+                session.close();
+            }
+        } catch (JMSException e) {
+            isExceptionThrown = true;
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        } finally {
+            try {
+                if (sessionCount.decrementAndGet() == 0) {
+                    connection.close();
+                }
+            } catch (JMSException e) {
+                if (!isExceptionThrown) {
+                    throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+                }
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public BytesMessage createBytesMessage() {
+        try {
+            return getSession().createBytesMessage();
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public MapMessage createMapMessage() {
+        try {
+            return getSession().createMapMessage();
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Message createMessage() {
+        try {
+            //TODO: Should create a Message object:currently creates a BytesMessage object
+            return getSession().createMessage();
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ObjectMessage createObjectMessage() {
+        try {
+            return getSession().createObjectMessage();
+        } catch (JMSException e) {
+            throw new JMSRuntimeException(e.getMessage(), e.getErrorCode(), e);
```

Comment:
This is required by the JMS 2.0 spec.
Quoting the specification (Section 10.2) - "In general, methods on interfaces defined in JMS 1.1 and earlier throw checked exceptions, whilst methods on the JMSContext, JMSProducer and JMSConsumer interfaces that were defined for the simplified API throw unchecked exceptions."